app/web/src/newhotness/ApplyChangeSetButton.vue

```diff
@@ -52,19 +52,29 @@ import {
 } from "@/workers/types/entity_kind_types";
 import { bifrost, useMakeArgs, useMakeKey } from "@/store/realtime/heimdall";
 import ApprovalFlowModal from "./ApprovalFlowModal.vue";
-import { assertIsDefined, Context } from "./types";
+import { useContext } from "./logic_composables/context";
 import { useApplyChangeSet } from "./logic_composables/change_set";
+import { useStatus } from "./logic_composables/status";
+import { ChangeSetStatus } from "@/api/sdf/dal/change_set";
 
-const ctx = inject<Context>("CONTEXT");
-assertIsDefined(ctx);
+const ctx = useContext();
 
 const approvalFlowModalRef = ref<InstanceType<typeof ApprovalFlowModal>>();
 
 const openApprovalFlowModal = () => {
   approvalFlowModalRef.value?.open();
 };
 
-const { applyChangeSet, disableApplyChangeSet } = useApplyChangeSet();
+const { loading, performApply: _performApply } = useApplyChangeSet(ctx);
+
+const status = useStatus();
+const disableButtonToOpenModal = computed(
+  () =>
+    (ctx.changeSet.value?.status !== ChangeSetStatus.Open &&
+      ctx.changeSet.value?.status !== ChangeSetStatus.NeedsApproval) ||
+    ctx.onHead.value ||
+    status.value === "syncing",
```

Comment:
Note: this now differs from the approval flow modal apply button. They were together and are now separate since approval state should not block the modal from being opened.